daemons/based/based_remote.c

```diff
@@ -441,11 +441,23 @@ cib_remote_msg(gpointer data)
     pcmk__client_t *client = ncd->new_client;
     xmlNode *command = NULL;
     int rc;
-    int timeout = 1000;
+    int timeout_sec = 1;
     const char *client_name = pcmk__client_name(client);
 
     if (pcmk_is_set(client->flags, pcmk__client_authenticated)) {
-        timeout = -1;
+        timeout_sec = 60;
+    }
+
+    /* If start_time is 0, we've previously handled a complete message and this
+     * callback is being reused for a new message.  Reset the start_time, giving
+     * this new message timeout_sec from now to complete.
+     */
+    if (ncd->start_time == 0) {
+        ncd->start_time = time(NULL);
+    } else if (time(NULL) >= ncd->start_time + timeout_sec) {
+        free(ncd);
+        crm_info("Error reading from remote client: %s", pcmk_rc_str(ETIME));
+        return -1;
```

Comment:
I'm not sure if this is at all necessary, or if I can just assume that every time the callback happens, we get a new ncd object and can free it when we're done with it.